webnn.json

```diff
@@ -0,0 +1,419 @@
+{
+  "error type": {
+    "category": "enum",
+    "values": [
+        {"value": 0, "name": "no error"},
+        {"value": 1, "name": "validation"},
+        {"value": 2, "name": "out of memory"},
+        {"value": 3, "name": "unknown"},
+        {"value": 4, "name": "device lost"}
+    ]
+  },
+  "char": {
+    "category": "native"
+  },
+  "void": {
+    "category": "native"
+  },
+  "const void*": {
+    "category": "native"
+  },
+  "uint32_t": {
+    "category": "native"
+  },
+  "const uint32_t*": {
+    "category": "native"
+  },
+  "const int32_t*": {
+    "category": "native"
+  },
+  "int32_t": {
+    "category": "native"
+  },
+  "size_t": {
+    "category": "native"
+  },
+  "bool": {
+    "category": "native"
+  },
+  "error filter": {
+    "category": "enum",
+    "values": [
+        {"value": 0, "name": "none"},
+        {"value": 1, "name": "validation"},
+        {"value": 2, "name": "out of memory"}
+    ]
+  },
+  "error callback": {
+    "category": "callback",
+    "args": [
+        {"name": "type", "type": "error type"},
+        {"name": "message", "type": "char", "annotation": "const*"},
+        {"name": "userdata", "type": "void", "annotation": "*"}
+    ]
+  },
+  "neural network context": {
+    "category": "object",
+    "methods": [
+      {
+        "name": "create model builder",
+        "returns": "model builder"
+      },
+      {
+        "name": "set uncaptured error callback",
+        "args": [
+            {"name": "callback", "type": "error callback"},
+            {"name": "userdata", "type": "void", "annotation": "*"}
+        ]
+      },
+      {
+          "name": "push error scope",
+          "args": [
+              {"name": "filter", "type": "error filter"}
+          ]
+      },
+      {
+          "name": "pop error scope",
+          "returns": "bool",
+          "args": [
+              {"name": "callback", "type": "error callback"},
+              {"name": "userdata", "type": "void", "annotation": "*"}
+          ]
+      }
+    ]
+  },
+  "operand type": {
+    "category": "enum",
+    "values": [
+      {"value": 0, "name": "float32"},
+      {"value": 1, "name": "float16"},
+      {"value": 2, "name": "int32"},
+      {"value": 3, "name": "uint32"}
+    ]
+  },
+  "operand descriptor": {
+    "category": "structure",
```

Comment:
Descriptor types should probably be extensible (not needed now, but important in the future to not break compatibility when adding extensions)

Reply:
ack that, will fix when adding extensions